Sources/Misc/Signing.swift

```diff
@@ -178,6 +187,23 @@ extension Signing {
 
 }
 
+extension Signing.SignatureParameters {
+
+    var asData: Data {
+        // Not modified responses sign the Etag
+        let payload: Data = self.notModifiedResponse
+        ? self.eTag?.asData ?? Data()
+        : self.message
```

Comment:
Given etag and message are exclusive, I would omit the etag from the signing parameters. For not modified responses the message IS the etag. That way you avoid having to have request-level logic in the signing code, gives you better abstraction IMHO. The signature for signature check should just be "local_nonce, server_timestamp, message, signature". And and leave the not-modified handling to the http request.

Reply:
Oh that's a great observation!
That was a nice small refactor, thanks: 59cdce772278473f458f357f883cb034629bf51d